backend/src/services/user-details/user-details.service.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  UserDetailsResponseDto,
+  UserDetailsUpdateInfoRequestDto,
+} from '~/common/types/types';
+import { userDetails as userDetailsRep } from '~/data/repositories/repositories';
+
+type Constructor = {
+  userDetailsRepository: typeof userDetailsRep;
+};
+
+class UserDetails {
+  #userDetailsRepository: typeof userDetailsRep;
+
+  public constructor({ userDetailsRepository }: Constructor) {
+    this.#userDetailsRepository = userDetailsRepository;
+  }
+
+  public async updateUserDetails(
+    userId: number,
+    userDetailsUpdateInfoRequestDto: UserDetailsUpdateInfoRequestDto,
+  ): Promise<UserDetailsResponseDto> {
+    const userDetailsByUserId = await this.#userDetailsRepository.getByUserId(
+      userId,
+    );
+
+    if (userDetailsByUserId) {
+      const userDetails = await this.#userDetailsRepository.updateUserDetails(
+        userDetailsByUserId.id,
+        userDetailsUpdateInfoRequestDto,
+      );
+
+      return userDetails;
+    }
+
+    const userDetails = await this.#userDetailsRepository.createUserDetails(
+      userId,
+      userDetailsUpdateInfoRequestDto,
+    );
```

Comment:
As an alternative - all this stuff could be achieved by on `upsert` method in `userDetailsRepository` - we have `user_id` col unique constraint so on insert with already existing row by `user_id` it would throw error that could be catched by `onConflict` - eg
```
  this.#UserDetailsModel
      .query()
      .insert({
        fullName,
        gender,
        dateOfBirth,
        userId,
      })
      .onConflict(['user_id'])
      .merge(userDetails)
      .execute();
```